Restructure top level folders

abhi-mosaic opened the pull request:
TODO:

- [x] READMEs for every top level folder
- [x] fix most tests
- future PR: fix tests that test tscripts (@dakinggg I could use your help)
- [x] fix any documentation and MCLI yamls whose paths have changed
- [x] look into if there is a cleaner presentation than having `assets`, `tests/`, bunch of other files at toplevel (@alextrott16)
- [x] ~figure out what to do about `data/finetuning/tasks.py`~

For best viewing experience, I recommend looking at the branch in Github UI: https://github.com/mosaicml/llm-foundry/tree/abhi/restruct

alextrott16 commented on README.md:
````diff
@@ -56,186 +46,48 @@ To get started, clone this repo and install the requirements:
 
 ```bash
 git clone https://github.com/mosaicml/llmfoundry.git
````
```suggestion
git clone https://github.com/mosaicml/llm-foundry.git
```

alextrott16 commented on README.md:
````diff
@@ -56,186 +46,48 @@ To get started, clone this repo and install the requirements:
 ```bash
 git clone https://github.com/mosaicml/llmfoundry.git
-pip install -e ".[gpu]"  # or pip install -e . if no NVIDIA GPU
 cd llmfoundry
````
```suggestion
cd llm-foundry
```

vchiley commented on train/README.md:
```diff
@@ -0,0 +1,292 @@
+You can read more about the benefits of using mosaicml-streaming [here](https://streaming.docs.mosaicml.com/en/stable/).
+
+NOTE: If you only want to profile these LLMs, we recommend that you **download and prepare the `train_small` and `val` splits**,
+and skip the full `train` split. You'll just need to replace `split: train` with `split: train_small` in your run YAML's dataloader config, [e.g. here](./llmfoundry/yamls/mosaic_gpt/125m.yaml).
```
```suggestion
and skip the full `train` split. You'll just need to replace `split: train` with `split: train_small` in your run YAML's dataloader config, [e.g. here](./llmfoundry/train/yamls/mosaic_gpt/125m.yaml).
```

vchiley commented on train/README.md:
```diff
@@ -0,0 +1,292 @@
+and stream the data to any compute cluster, with any number of devices, and any number of CPU workers, and it all ~ just works ~ .
+You can read more about the benefits of using mosaicml-streaming [here](https://streaming.docs.mosaicml.com/en/stable/).
+
+NOTE: If you only want to profile these LLMs, we recommend that you **download and prepare the `train_small` and `val` splits**,
```
```suggestion
NOTE: If you only want to profile these LLMs, we recommend that you **download and prepare the `train_small` and `val_small` splits**,
```

Also requires an update [here](https://github.com/mosaicml/llm-foundry/blob/abhi/restruct/data_prep/convert_dataset.py#L39)
(we have val_small, but didn't add it as an option)

vchiley commented on train/README.md:
```diff
@@ -0,0 +1,292 @@
+
+NOTE: If you only want to profile these LLMs, we recommend that you **download and prepare the `train_small` and `val` splits**,
+and skip the full `train` split. You'll just need to replace `split: train` with `split: train_small` in your run YAML's dataloader config, [e.g. here](./llmfoundry/yamls/mosaic_gpt/125m.yaml).
+You can also accomplish this in your CLI command like so: `composer train.py ... train_loader.dataset.split=train_small`
```
```suggestion
You can also accomplish this in your CLI command like so: `composer train.py ... train_loader.dataset.split=train_small eval_loader.dataset.split=val_small`
```

abhi-mosaic commented:
Moved scripts to a `scripts/` folder. Fixed a lot of references but probably not all of them. Will leave to tomorrow's bug bash (and relevant owners) to fix READMEs for each subfolder

vchiley reviewed (approved):
❤️

abhi-mosaic commented:
Merged as it's getting late, and the pytest-cpu tests are all passing, just something wrong with `tests/conftest.py`
